tests/pytests/unit/cloud/clouds/test_hetzner.py

```diff
@@ -411,6 +444,36 @@ def test_create_networks(vm):
         assert args.kwargs["networks"] == ["a", "c"]
 
 
+class Timeout:
+    def __init__(self, seconds=1, error_message="Timeout"):
+        self.seconds = seconds
+        self.error_message = error_message
+
+    def handle_timeout(self, signum, frame):
+        raise TimeoutError(self.error_message)
+
+    def __enter__(self):
+        signal.signal(signal.SIGALRM, self.handle_timeout)
+        signal.alarm(self.seconds)
+
+    def __exit__(self, type, value, traceback):
+        signal.alarm(0)
+
+
+def test_create_timeout_when_server_not_running(vm):
+    with patch(
+        "salt.cloud.clouds.hetzner._connect_client", return_value=MagicMock()
+    ) as connect:
+        # Create waits for the server to be running before going on
+        created_server = MagicMock()
+        created_server.status = "initializing"
+        connect.return_value.servers.get_by_id.return_value = created_server
+
+        with pytest.raises(TimeoutError):
+            with Timeout(seconds=3):
+                hetzner.create(vm)
```

Comment:
**suggestion (non-blocking)** it would be helpful to mention something either in the comments or in the actual test name. I would say something like:

```
def test_when_server_status_is_not_running_create_should_never_return(vm):
```

Alternatively the comment could be

> Create will never return unless the .status is "running", so we need a Timeout that will interrupt the loop in `.create`.

I'm not currently familiar with the surrounding code, so I'm assuming that the cloud code will already timeout this call?